### DISCUSSION TITLE
Is there any documentation for making bottlerocket work without the internet access to the instances?

### DISCUSSION THREAD

**Discussion by soura49** [bottlerocket-os/bottlerocket#3953] [General]
* We are using Bottlerocket AMI for EKS-managed Node groups
* Right Now, We have Egress for Internet open from Node Security Group 
* But when we Remove that it fails to join the cluster and load the kernel modules etc.
* Is there a list of Internet Calls that Bottlerocket AMI does for starting up?

**Comment by jpculp** [comment]
I got this working in the past with AWS PrivateLink by adding the following regional VPC endpoints:

**ECR** _(even if you use a different registry it's necessary for host containers and pause container)_
- com.amazonaws.us-west-2.s3 (gateway) _# required to pull the ECR image layers_
- com.amazonaws.us-west-2.api
- com.amazonaws.us-west-2.ecr.dkr

**EKS**
- com.amazonaws.us-west-2.eks
- com.amazonaws.us-west-2.sts _# required when using IAM roles for service accounts_

**IMDS** _(necessary for populating settings on first boot)_
- com.amazonaws.us-west-2.ec2
- com.amazonaws.us-west-2.ec2messages

**Bottlerocket control container**
- com.amazonaws.us-west-2.ssm
- com.amazonaws.us-west-2.ssmmessages

**Reply by soura49** [reply]
Thank you will make these changes and will test it out.

**Comment by soura49** [comment]
- Thank you @jpculp
- I made progress on the things.
- Only thing that is failing right now is that it is in NotReady State foreever.
- Are there any other API calls that are required but missing?

**Reply by jpculp** [reply]
Are these fresh nodes from after the endpoints were created? If not they might be missing some settings that they couldn't populate on first boot. I would try spinning down the cluster and then spinning back up with new nodes.

**Comment by soura49** [comment]
* These are Fresh Nodes i created a new managed nodegroup after making the changes and they are still in NotReady State.
* I found this document https://repost.aws/knowledge-center/eks-cluster-node-group-private-network which mentions creating the VPC endpoint for STS and cloud watch logs along with the ones that you have mentioned above.
*  I am concerned only about the STS do we need the STS VPCendpoint as well?

**Reply by larvacea** [reply]
[EKS private clusters](https://docs.aws.amazon.com/eks/latest/userguide/private-clusters.html) says that an STS endpoint is required if your pod uses IAM Roles for service accounts.

**Reply by soura49** [reply]
* The exact issue was `aws-vpc-cni` was not healthy because of the IAM role for Service Account that is making it `NotReady`.
* Once the STS Vpc endpoint is made it is healthy.

**Reply by larvacea** [reply]
Thanks for the update! It's great to hear what unblocked you.

**Reply by jpculp** [reply]
Awesome find! I added it to the list above.
